### ISSUE TITLE
On Bundler 2.5 the `file: '.ruby-version'` dsl doesn't work anymore, fails with "Undefined local variable or method `gemfile' for Gemfile. Bundler cannot continue." 

### ISSUE THREAD

**Issue by AndrewSwerlick** [rubygems/rubygems#7286]
### Describe the problem as clearly as you can

After upgrading to bundler 2.5 we're seeing multiple binstubs are now failing with the error 
```
[!] There was an error parsing `Gemfile`: Undefined local variable or method `gemfile' for Gemfile. Bundler cannot continue.
```

We've traced the issue to the following line 
https://github.com/rubygems/rubygems/blob/d66877d87a53515523040f22ec6971f38a6d5339/bundler/lib/bundler/ruby_dsl.rb#L44

It seems that the Gemfile syntax 
```
ruby file: <some_filename>
```
no longer works and triggers this missing method error. 

This pull request seems like a potential culprit https://github.com/rubygems/rubygems/pull/7250/files

It appears the mock DSL setting up for testing masked this problem in tests, but it shows up in real world usage

### Post steps to reproduce the problem

In a local project upgrade to bundler 2.50 and attempt to run any binstub.

### Which command did you run?
We tried both `bin/vite`, `bin/rubocop`, and `bin/rails` as test cases and all failed with the same issue

### What were you expecting to happen?

For the binstubs to run successfully

### What actually happened?

```
[!] There was an error parsing `Gemfile`: Undefined local variable or method `gemfile' for Gemfile. Bundler cannot continue.

 #  from <redacted>/Gemfile:3
 #  -------------------------------------------
 #  
 >  ruby file: '.ruby-version'
 #  
 #  -------------------------------------------
```

### If not included with the output of your command, run `bundle env` and paste the output below
Fails with the same error
```
$ bundle env
[!] There was an error parsing `Gemfile`: Undefined local variable or method `gemfile' for Gemfile. Bundler cannot continue.

 #  from <redacted>/Gemfile:3
 #  -------------------------------------------
 #  
 >  ruby file: '.ruby-version'
 #  
 #  -------------------------------------------
``
<!-- Replace this with the result of `bundle env`. Don't forget to anonymize any private data! -->

**Comment by deivid-rodriguez** [comment]
Thanks for the report, let me look into this!

**Comment by AndrewSwerlick** [comment]
> Thanks for the report, let me look into this!

Awesome thanks. Worth noting that we found that just adding an `attr_reader :gemfile` to that `ruby_dsl` file appeared to fix the issue, although we didn't test extensively.

**Comment by deivid-rodriguez** [comment]
Thanks @AndrewSwerlick, looking into the original issue we intended to fix, adding an `attribute_reader :gemfile` is exactly the right solution. Working on the fix now, will add you as co-author there and push a new version ASAP :)

**Comment by martinemde** [comment]
@AndrewSwerlick Thanks for finding and reporting this! This was my mistake. The RubyDsl tests are too damn mocked!

**Comment by deivid-rodriguez** [comment]
My fault too 🤚, since I review and merged the culprit PR and failed to notice the missing reader!

**Comment by deivid-rodriguez** [comment]
We just pushed a fix for this, let us know if Bundler 2.5.1 is fine!

**Comment by bkuhlmann** [comment]
Thanks and I've confirmed that using `rubygems-update` and `bundler` 2.5.1, locally, appears to have resolved the issue.
